src/providers/codeforces.cr

```diff
@@ -0,0 +1,37 @@
+class Codeforces < SiteStats
+  @@url_regex = Regex.new("https?://codeforces[.]com/profile/(?<username>\\w+)")
+  # https://codeforces.com/apiHelp
+  @@api_url = "https://codeforces.com/api/user.status?handle=%s&from=1&count=10000"
+
+  def self.fetch_lang_stats(username : String) : Hash(Langname, Int32)?
+    stats = Hash(Langname, Int32).new(0)
+
+    url = URI.parse(@@api_url % username)
+    response = HTTP::Client.get(url)
+    return nil unless response.success?
+
+    obj = JSON.parse(response.body)
+    return nil if obj.nil? || obj["status"] != "OK"
+
+    obj["result"].as_a.each do |item|
+      next if item["verdict"] != "OK"
+
+      language = item["programmingLanguage"].as_s
+      # Removes espaces for the JSON to work later on:
+      #     PyPy 3-64 -> PyPy-3-64
+      language = language.gsub(/ /, "-")
```

Comment:
This should not be necessary. I would add `"pypy364" => "Python"` (along with other possible PyPys) to the `@@lang_aliases` in site-stats.cr instead.

Reply:
If espaces are not removed you get the following error:

DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided ('PyPy 3-64') contains HTML space characters, which are not valid in tokens.
    at renderStatsTable (http://localhost:8080/main.js:64:23)
    at http://localhost:8080/main.js:101:13